mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/query/parsing/nativequery/impl/MongoDBQueryDescriptorBuilder.java

```diff
@@ -100,4 +152,37 @@ public DBObject parse(String json) {
 		return (DBObject) object.get( "json" );
 	}
 
+	private static DBObject stage(String key, Object value) {
+		DBObject stage = new BasicDBObject();
+		stage.put( key, value );
+		return stage;
+	}
+
+	public boolean addPipeline(StackElement element, String value) {
+		if ( !parsed.contains( element.getIndex() ) ) {
+			parsed.add( element.getIndex() );
+			boolean isJsonObject = value.trim().matches( "^\\{.*:.*\\}$" );
```

Comment:
Missed that.

I think we should rename the existing one `parseDBObject` and have the `parse` method returning an object. `parseDBObject` would call `parse` and cast the result.

WDYT?